src/Compilers/Core/Portable/DiagnosticAnalyzer/CompilationWithAnalyzers.cs

```diff
@@ -121,7 +121,13 @@ public CompilationWithAnalyzers(Compilation compilation, ImmutableArray<Diagnost
         {
         }
 
-        private CompilationWithAnalyzers(Compilation compilation, ImmutableArray<DiagnosticAnalyzer> analyzers, CompilationWithAnalyzersOptions analysisOptions, CancellationToken cancellationToken)
+        /// <inheritdoc cref="CompilationWithAnalyzers(Compilation, ImmutableArray{DiagnosticAnalyzer}, CompilationWithAnalyzersOptions)"/>
+        /// <param name="cancellationToken">A cancellation token that can be used to abort analysis.</param>
+        public CompilationWithAnalyzers(
```

Comment:
This adds a public API. Why isn't CI requiring this to be registered in the publicAPI tracking files?
Also, public APIs need explicit validation.